packages-exp/auth-exp/src/core/user/token_manager.test.ts

```diff
@@ -187,6 +189,97 @@ describe('core/user/token_manager', () => {
       const tokens = (await stsTokenManager.getToken(auth))!;
       expect(tokens).to.eql('token');
     });
+
+    describe('in passthrough mode', () => {
+      const fakeCustomToken = 'fake-custom-token';
+      const provider = {
+        async getCustomToken(): Promise<string> {
+          return fakeCustomToken;
+        }
+      };
+
+      it('refreshes the token if forceRefresh is true and custom token provider is set', async () => {
+        Object.assign(stsTokenManager, {
+          accessToken: 'old-access-token',
+          expirationTime: now + 100_000,
+          isPassthroughMode: true
+        });
+        setCustomTokenProvider(auth, provider);
+        stub(auth, '_refreshWithCustomTokenProvider').returns(
+          Promise.resolve('new-access-token')
+        );
+
+        const tokens = (await stsTokenManager.getToken(auth, true))!;
+
+        expect(tokens).to.eq('new-access-token');
```

Comment:
Nit: Noticed that I'm using eq() and eql() interchangeably for string comparisons. I don't think it matters in this case but I'll update the tests for token_manager to be more consistent in the next revision

Reply:
Yeah it doesn't really matter for strings, don't worry too much if it's not always consistent